exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/_metric_exporter/__init__.py

```diff
@@ -79,103 +80,143 @@ def __init__(
         )
 
     def _translate_data(
-        self, data: Sequence[Metric]
+        self, data: MetricsData
     ) -> ExportMetricsServiceRequest:
-        sdk_resource_scope_metrics = {}
-
-        for metric in data:
-            resource = metric.resource
-            scope_map = sdk_resource_scope_metrics.get(resource, {})
-            if not scope_map:
-                sdk_resource_scope_metrics[resource] = scope_map
-
-            scope_metrics = scope_map.get(metric.instrumentation_scope)
-
-            if not scope_metrics:
-                if metric.instrumentation_scope is not None:
-                    scope_map[metric.instrumentation_scope] = pb2.ScopeMetrics(
-                        scope=InstrumentationScope(
-                            name=metric.instrumentation_scope.name,
-                            version=metric.instrumentation_scope.version,
-                        )
-                    )
-                else:
-                    scope_map[
-                        metric.instrumentation_scope
-                    ] = pb2.ScopeMetrics()
 
-            scope_metrics = scope_map.get(metric.instrumentation_scope)
+        resource_instrumentation_scope_pb2_scope_metrics = {}
```

Comment:
It really helped me to write this code, as the dictionaries are being traversed it helps to know in the name the type of the keys and values that are nested within it. We could use other name but this one is exact and descriptive, I had difficulties using other names because it was hard to keep track of the dictionary components as they were being traversed.